src/nbl/ui/CWindowWin32.cpp

```diff
@@ -8,276 +8,70 @@ namespace ui
-			{0, WM_MBUTTONDOWN, nbl::EMIE_MMOUSE_PRESSED_DOWN},
-			{1, WM_MBUTTONUP,   nbl::EMIE_MMOUSE_LEFT_UP},
-			{2, WM_MOUSEMOVE,   nbl::EMIE_MOUSE_MOVED},
-			{3, WM_MOUSEWHEEL,  nbl::EMIE_MOUSE_WHEEL},
-			{-1, 0, 0}
-		};
-
-		// handle grouped events
-		messageMap* m = mouseMap;
-		while (m->group >= 0 && m->winMessage != message)
-			m += 1;
-
-		if (m->group >= 0)
-		{
-			if (m->group == 0)	// down
+			if (wParam = TRUE)
 			{
-				ClickCount++;
-				SetCapture(hWnd);
+				eventCallback->onWindowShown(window);
 			}
 			else
-				if (m->group == 1)	// up
-				{
-					ClickCount--;
-					if (ClickCount < 1)
-					{
-						ClickCount = 0;
-						ReleaseCapture();
-					}
-				}
-
-			event.EventType = nbl::EET_MOUSE_INPUT_EVENT;
-			event.MouseInput.Event = (nbl::EMOUSE_INPUT_EVENT) m->irrMessage;
-			event.MouseInput.X = (short)LOWORD(lParam);
-			event.MouseInput.Y = (short)HIWORD(lParam);
-			event.MouseInput.Shift = ((LOWORD(wParam) & MK_SHIFT) != 0);
-			event.MouseInput.Control = ((LOWORD(wParam) & MK_CONTROL) != 0);
-			// left and right mouse buttons
-			event.MouseInput.ButtonStates = wParam & (MK_LBUTTON | MK_RBUTTON);
-			// middle and extra buttons
-			if (wParam & MK_MBUTTON)
-				event.MouseInput.ButtonStates |= nbl::EMBSM_MIDDLE;
-#if(_WIN32_WINNT >= 0x0500)
-			if (wParam & MK_XBUTTON1)
-				event.MouseInput.ButtonStates |= nbl::EMBSM_EXTRA1;
-			if (wParam & MK_XBUTTON2)
-				event.MouseInput.ButtonStates |= nbl::EMBSM_EXTRA2;
-#endif
-			event.MouseInput.Wheel = 0.f;
-
-			// wheel
-			if (m->group == 3)
 			{
-				POINT p; // fixed by jox
-				p.x = 0; p.y = 0;
-				ClientToScreen(hWnd, &p);
-				event.MouseInput.X -= p.x;
-				event.MouseInput.Y -= p.y;
-				event.MouseInput.Wheel = ((float)((short)HIWORD(wParam))) / (float)WHEEL_DELTA;
-			}
-
-			dev = getDeviceFromHWnd(hWnd);
-			if (dev)
-			{
-				dev->postEventFromUser(event);
-
-				if (event.MouseInput.Event >= nbl::EMIE_LMOUSE_PRESSED_DOWN && event.MouseInput.Event <= nbl::EMIE_MMOUSE_PRESSED_DOWN)
-				{
-					uint32_t clicks = dev->checkSuccessiveClicks(event.MouseInput.X, event.MouseInput.Y, event.MouseInput.Event);
-					if (clicks == 2)
-					{
-						event.MouseInput.Event = (nbl::EMOUSE_INPUT_EVENT)(nbl::EMIE_LMOUSE_DOUBLE_CLICK + event.MouseInput.Event - nbl::EMIE_LMOUSE_PRESSED_DOWN);
-						dev->postEventFromUser(event);
-					}
-					else if (clicks == 3)
-					{
-						event.MouseInput.Event = (nbl::EMOUSE_INPUT_EVENT)(nbl::EMIE_LMOUSE_TRIPLE_CLICK + event.MouseInput.Event - nbl::EMIE_LMOUSE_PRESSED_DOWN);
-						dev->postEventFromUser(event);
-					}
-				}
+				eventCallback->onWindowHidden(window);
 			}
-			return 0;
+			break;
 		}
-
-		switch (message)
-		{
-		case WM_PAINT:
+		case WM_MOVING: [[fallthrough]]
+		case WM_MOVE:
 		{
-			PAINTSTRUCT ps;
-			BeginPaint(hWnd, &ps);
-			EndPaint(hWnd, &ps);
+			uint32_t newX = (int)LOWORD(lParam); 
+			uint32_t newY = (int)HIWORD(lParam);
+			eventCallback->onWindowMoved(window, newX, newY);
```

Comment:
careful... you need to use `int32_t` here.. and probably even change the signature of `onWindowMoved`